src/components/containers/TaskList.js

```diff
@@ -7,105 +7,147 @@ import { TaskSection } from '../presentational/Dropdown';
 
 import TaskService from '../../services/TaskService';
 
-const initTasks = [...Array(5).keys()].map((num) => ({
-  id: num,
-  name: 'Placeholder Title',
-  scheduledDate: 'Tues 3:00 pm - 4:00 pm',
-  duration: 90,
-  dueDate: 'Tue Dec 10 2019',
-  category: 'Testing',
-  notes: 'Lorem ipsum',
-  created: '10/10/19',
-  scheduled: false
-}));
-
-const scheduledTasks = initTasks.map((task) => ({ ...task, scheduled: true }));
-
 const emptyTask = {
   id: 0,
-  title: '',
-  scheduledDate: '',
+  name: '',
   duration: 90,
-  dueDate: new Date().toDateString(),
+  description: '',
   category: '',
-  notes: '',
-  created: ''
+  created: '',
+  scheduledDate: '',
+  dueDate: new Date().toDateString()
 };
 
-const categories = ['School', 'Work', 'Personal', 'Holidays'];
+const categories = ['School', 'Work', 'Personal', 'Holidays']; // temporary
 
 const TaskList = () => {
-  // EXAMPLE CREATE AND RETRIEVE TASK
-  let time_now = new Date().toISOString()
-  TaskService.postTask("Homework", "Alex", 1300, time_now, function (task) {
-    console.log("Task Posted")
-    console.log(task)
-
-    TaskService.getTask(1, function (task) {
-      console.log("Task Retrieved")
-      console.log(task)
-    })
-  })
-
-  // TaskService.getTask(1, function (task) {
-  //   console.log("Task Retrieved")
-  //   console.log(task)
-  // })
-
-  const [tasks, setTasks] = useState(initTasks);
-  const [schedTasks, setSchedTasks] = useState(scheduledTasks);
+  /* Sample TaskService use: */
+  /* let time_now = new Date().toISOString();
+  TaskService.postTask('Homework', 'Alex', 1300, function(task) {
+    console.log('Task Posted');
+    console.log(task);
+
+    TaskService.getTask(1, function(task) {
+      console.log('Task Retrieved');
+      console.log(task);
+    });
+  }); */
+
+  const [tasks, setTasks] = useState([]);
+  const [unscheduled, setUnscheduled] = useState([]);
+  const [scheduled, setScheduled] = useState([]);
+
   const [creating, setCreating] = useState(false);
   const [selected, setSelected] = useState([]);
 
+  const splitTasks = () => {
+    let schedArr = [],
+      unschedArr = [];
+    tasks.forEach((task) => {
+      if (task.scheduled) {
+        schedArr.push(task);
+      } else {
+        unschedArr.push(task);
+      }
+    });
+    setUnscheduled(unschedArr);
+    setScheduled(schedArr);
+  };
+
+  useEffect(() => {
+    /* split into scheduled and unscheduled every time tasks updates */
+    splitTasks();
+  }, [tasks]);
+
+  const createTask = (task) => {
+    const { name, description, duration, due } = task;
+    TaskService.postTask(name, description, duration, due, (task) => {
+      console.log('Creating:', task);
+      setTasks(
+        tasks.concat({
+          /* temporarily filling out extra fields */
+          ...task,
+          created: '10/10/10',
+          category: 'School',
+          dueDate: 'Tue Dec 10 2019',
+          scheduledDate: 'Tues 3-4'
+        })
```

Comment:
Why are we filling these out? The task returned should have this info (as of a recent backend change). Can we default to what the returned task says?

Reply:
Also, please send the user-specified due date to the backend when creating

Reply:
Updated to match database models.